src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,48 +132,59 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            _children.Enqueue(result);
+#else
+            _children.Add(result);
+#endif
 
```

Comment:
Sorry, I don't understand your comment. It seems natural to use the _children field internally while external clients use the Children property on the interface. External clients don't have or need access to adding/enqueuing results.

Is this commit an experiment?